### PR TITLE
refactor: add more clippy rule

### PR DESCRIPTION
**Pull request by hardfist**
## Summary
related to #7589

## Checklist

- [ ] Tests updated (or not required).
- [ ] Documentation updated (or not required).

### CODE REVIEW

File: crates/rspack_core/src/compiler/module_executor/execute.rs [hunk 1]
```diff
@@ -269,7 +267,7 @@ impl Task<MakeTaskContext> for ExecuteTask {
 
         result.id = id;
 
-        modules.iter().for_each(|m| {
+        for m in modules.iter() {
```
Comment (hardfist) [inline crates/rspack_core/src/compiler/module_executor/execute.rs:270]: I'm not sure this is better?

File: crates/rspack_core/src/concatenated_module.rs [hunk 2]
```diff
@@ -409,7 +409,7 @@ impl ConcatenatedModule {
     for m in modules {
       identifiers.push(m.shorten_id.as_str());
     }
-    identifiers.sort();
+    identifiers.sort_unstable();
```
Comment (hardfist) [inline crates/rspack_core/src/concatenated_module.rs:412]: sort_unstable seems faster than sort?

**Comment by hardfist** [comment]
!bench

**Comment by h-a-n-a** [comment]
Would you please fix the conflict and let's get it merged first?

**Review by h-a-n-a — COMMENTED** [review]
Let's ship this first

**Review by h-a-n-a — APPROVED** [review]

**Pull request merged**